vendor/bat-native-ads/src/bat/ads/internal/ad_targeting/processors/contextual/text_classification/text_classification_processor.cc

```diff
@@ -48,6 +63,20 @@ void TextClassification::Process(const std::string& text) {
     return;
   }
 
+  const chrome_lang_id::NNetLanguageIdentifier::Result res_id = detect_page_language(text);
+
+  const std::string locale = brave_l10n::LocaleHelper::GetInstance()->GetLocale();
+  const std::string locale_language = brave_l10n::GetLanguageCode(locale);
+
+  // NOTE: This holds as long as the resource is univocally matched to the locale 
+  if (res_id.is_reliable && locale_language != res_id.language) {

```

Comment:
so this means that if we have low confidence in the detection, we classify? Why not `if (!res_id.is_reliable || locale_language != res_id.language)`?

Reply:
My thinking was: if the model has low confidence then disregard the model and go with default behaviour. But it's true that even unreliability of the detection is some actionable information. If we trust `cld_3` your condition is probably best.